### PR TITLE
[feat] add CPU support to tutorials in examples + factorize tutorials

### PR DESCRIPTION
**Pull request by jessijzhao**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
* adds CPU support to tutorials in examples --> works on machine without CUDA
* factorizes tutorials in examples to remove duplicate code
* fixes minor typos

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: examples/tutorial_oss.py [hunk 1]
```diff
@@ -4,45 +4,25 @@
...
 def train(rank: int, world_size: int, epochs: int, use_oss: bool):
 
     # DDP
     dist_init(rank, world_size)
+    rank = torch.device("cpu") if DEVICE == "cpu" else rank
 
     # Problem statement
     model = getModel().to(rank)
-    dataloader = getData()
+    dataloader = [getData()]
```
Comment (blefaudeux) [inline examples/tutorial_oss.py:25]: hmm the name was not super explicit here, more `getDataloader`, but it's a bit more typical to get an iterator here than just a [list], no ? Else the loop down below will not go very far..

**Review by blefaudeux — APPROVED** [review]
Looks good to me, except for the data loader change (see comment), unless I missed something ? If I'm not mistaken and using a dataloader is simpler (if), would you mind updating the PR ? Thanks in any case @jessijzhao !

**Review by min-xu-ai — COMMENTED** [review]
Looks really good to me. Just curious, are you going to add an example of mixing pipe and oss or other zero features together?

**Comment by jessijzhao** [comment]
Thanks for the feedback! @min-xu-ai I might be able to add it next week, I'll keep you updated!

**Comment by blefaudeux** [comment]
Thank you @jessijzhao, very much appreciated !

**Pull request merged**